specs/switch.md

```diff
@@ -0,0 +1,117 @@
+# fast-switch
+
+## Overview
+
+*Switch* mimics a physical switch that can be turned **on** or **off**.
+
+### Use Cases
+
+Typical use cases include, but are not limited to, turning a feature on and off or showing or hiding a piece of UI
+  
+### Features
+
+A switch should allow the following properties:
+- `checked`, whether the switch is cheched or not.
+- `checked-message`, the message that displays when the switch is checked.
+- `unchecked-message`, the message that displays when switch is not checked.
+- `disbaled`, whether the switch is disabled or not.
+
+### Risks and Challenges
+
+Typically it's best to include a checked and unchecked message but it is not required. It's common to see *Switch* without a message but there are accessibility concerns since it's not necessarily obvious what is **on** and what is **off**. The visual representation of **on** tends to show the state indicator to the right with the background filled and **off** tends to show the state indicator to the left and the background unfilled or as a muted or neutral color. However, this pattern is not always consistent.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/toggle)
+- [Material UI](https://material-ui.com/components/switches/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/checkbox-toggle/)
+- [Ant Design](https://ant.design/components/switch/)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/toggle)
+- [Open UI](https://open-ui.org/components/switch)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/toggles)
+
+---
+
+## Design
+
+*Describe the design of the component, thinking through several perspectives:*
+
+- *A customer using the component on a web page.*
+- *A developer building an app with the component and interacting through HTML/CSS/JavaScript.*
+- *A designer customizing the component.*
+
+### API
+
+*Component name:*
+- `fast-switch`
+
+*Attributes:*
+- `checked`: boolean
```

Comment:
It might be good to enumerate slots separately since they're not really an attribute and conceptually they're pretty different